lnrpc/routerrpc/router_server.go

```diff
@@ -356,20 +360,62 @@ func (s *Server) SendPaymentV2(req *SendPaymentRequest,
 	)
 }
 
-// EstimateRouteFee allows callers to obtain a lower bound w.r.t how much it
-// may cost to send an HTLC to the target end destination.
+// EstimateRouteFee allows callers to obtain an upper bound w.r.t how much it
```

Comment:
maybe I'm being a bit picky here, but how is the result guaranteed to be an upper bound? it's definitely not a lower bound (like query routes might be) but you can probably pay more if network conditions change between probe and attempt. imo it's an expected value.

Reply:
Fair, I was considering it an upper bound as we select the most expensive hop hint in case there are multiple route hints to choose from. But since the overall result depends on dynamic fee changes along a route I'll go with your suggestion.